jobsdb/jobsdb.go

```diff
@@ -1240,37 +1233,26 @@ func mapDSToLevel(ds dataSetT) (levelInt int, levelVals []int, err error) {
 	return len(levelVals), levelVals, nil
 }
 
-func (jd *HandleT) createTableNames(dsIdx string) (string, string) {
+func (jd *HandleT) newDataSetStruct(dsIdx string) dataSetT {
 	jobTable := fmt.Sprintf("%s_jobs_%s", jd.tablePrefix, dsIdx)
 	jobStatusTable := fmt.Sprintf("%s_job_status_%s", jd.tablePrefix, dsIdx)
-	return jobTable, jobStatusTable
+	return dataSetT{
+		JobTable:       jobTable,
+		JobStatusTable: jobStatusTable,
+		Index:          dsIdx,
+	}
 }
 
-func (jd *HandleT) addNewDS(newDSType string, insertBeforeDS dataSetT) dataSetT {
-	jd.logger.Infof("Creating new DS of type %s before ds %s for %s jobsdb", newDSType, insertBeforeDS.Index, jd.tablePrefix)
-	var newDSIdx string
-	appendLast := newDSType == appendToDsList
+func (jd *HandleT) addDS(ds dataSetT, isNew bool) {
```

Comment:
I don't like the fact that we are passing the isNew bool, it makes things less readable and harder to follow.

Could we have a better approach, for example to different methods:
* `addDS(ds dataSetT)`
* `addNewDs(ds dataSetT)`